nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/CaptureChangeMySQL.java

```diff
@@ -0,0 +1,928 @@
+                case QUERY:
+                    // Is this the start of a transaction?
+                    QueryEventData queryEventData = event.getData();
+                    String sql = queryEventData.getSql();
+                    if ("BEGIN".equals(sql)) {
+                        // If we're already in a transaction, something bad happened, alert the user
+                        if (inTransaction) {
+                            throw new IOException("BEGIN event received while already processing a transaction. This could indicate that your binlog position is invalid.");
+                        }
+                        // Mark the current binlog position in case we have to rollback the transaction (if the processor is stopped, e.g.)
+                        xactBinlogFile = currentBinlogFile;
+                        xactBinlogPosition = currentBinlogPosition;
+                        xactSequenceId = currentSequenceId.get();
+
+                        BeginTransactionEventInfo beginEvent = new BeginTransactionEventInfo(timestamp, currentBinlogFile, currentBinlogPosition);
+                        currentSequenceId.set(beginEventWriter.writeEvent(currentSession, beginEvent, currentSequenceId.get()));
+                        inTransaction = true;
+                    } else if ("COMMIT".equals(sql)) {
+                        if (!inTransaction) {
+                            throw new IOException("COMMIT event received while not processing a transaction (i.e. no corresponding BEGIN event). "
+                                    + "This could indicate that your binlog position is invalid.");
+                        }
+                        // InnoDB generates XID events for "commit", but MyISAM generates Query events with "COMMIT", so handle that here
+                        CommitTransactionEventInfo commitTransactionEvent = new CommitTransactionEventInfo(timestamp, currentBinlogFile, currentBinlogPosition);
+                        currentSequenceId.set(commitEventWriter.writeEvent(currentSession, commitTransactionEvent, currentSequenceId.get()));
+                        // Commit the NiFi session
+                        session.commit();
+                        inTransaction = false;
+                        currentTable = null;
+
+                    } else {
+                        // Check for schema change events (alter table, e.g.). Normalize the query to do string matching on the type of change
+                        String normalizedQuery = sql.toLowerCase().trim().replaceAll(" {2,}", " ");
+                        if (normalizedQuery.startsWith("alter table")
+                                || normalizedQuery.startsWith("alter ignore table")
+                                || normalizedQuery.startsWith("drop table")) {
+                            SchemaChangeEventInfo schemaChangeEvent = new SchemaChangeEventInfo(currentTable, timestamp, currentBinlogFile, currentBinlogPosition, normalizedQuery);
```

Comment:
When I execute `drop database ...`, a binlog event reached here but got ignored. We probably should clear cache entries with drop database as well.